### PR TITLE
Announcement screen enhancement by adding upload image element 

### PR DESCRIPTION
**Pull request by mohagpt**
This PR introduces the QuickFixUploadImageSheet, a new reusable component that allows users to upload images in two ways: by taking a picture using the camera or selecting an image from their gallery.
Capture d'écran 2024-11-10 215719
Related to #131 

The feature was initially implemented as part of the upload-image feature and has now been tested to ensure its functionality.

Key points:
- Enhanced Announcement Screen: The QuickFixUploadImageSheet is integrated into the Announcement screen, allowing users to upload images easily when creating or editing announcements.
- Reusable Component: This element is designed to be reusable across the QuickFix app, with future plans to integrate it in other parts of the app, such as the profile screen, where users can upload their profile pictures.

Commit History:
- Resolves a bug caused by the previous feature implementation of the bottom bar adaptability to the navigationActions.
- Adds the reusable image upload feature with camera and gallery options.
- test: Adds tests for the QuickFixUploadImage component.
- 
This update improves the user experience and adds more flexibility for image uploads throughout the app.

### CODE REVIEW

File: app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixUploadImage.kt [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                            .padding(vertical = 8.dp)
+                            .testTag("libraryRow"),
+                    verticalAlignment = Alignment.CenterVertically,
+                    horizontalArrangement = Arrangement.Start) {
+                      Icon(
+                          painter = painterResource(id = R.drawable.upload_image),
+                          contentDescription = "Choose from library",
+                          tint = colorScheme.onBackground,
+                          modifier = Modifier.padding(start = 16.dp))
+                      Spacer(modifier = Modifier.width(108.dp))
```
Comment (daferya) [inline app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixUploadImage.kt:100]: It would be better to replace Modifier.width(108.dp) with Modifier.weight() for the spacing, as this absolute width may not display correctly on all phones. Using weights will make the layout more adaptable and ensure consistent spacing across different screen sizes.

File: app/src/main/java/com/arygm/quickfix/ui/search/Announcement.kt
Comment (daferya) [inline app/src/main/java/com/arygm/quickfix/ui/search/Announcement.kt:1]: You did a gread job. The layout adjustments are consistent with the existing design, ensuring a cohesive user experience.

File: app/src/androidTest/java/com/arygm/quickfix/ui/elements/QuickFixUploadImageTest.kt
Comment (daferya) [inline app/src/androidTest/java/com/arygm/quickfix/ui/elements/QuickFixUploadImageTest.kt:1]: Great job on this! You covered the main scenarios well, and the implementation looks solid. For next time, try to include tests for edge cases as well .This will help make the feature even more robust and handle unexpected situations . Keep up the good work!

File: app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixUploadImage.kt
Comment (daferya) [inline app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixUploadImage.kt:1]: Great job on this! The implementation looks clean and effective. Just a suggestion for next time: try adding some comments to explain key parts of the code. This will make it easier for others (and yourself) to understand the logic at a glance.

**Review by daferya — APPROVED** [review]
Great job on this! The implementation is clean and effective, with thoughtful layout adjustments that maintain a cohesive user experience. You covered the main scenarios well, and the code looks solid overall.

File: app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixUploadImage.kt (hunk 1 above)
Comment (mohagpt) [reply app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixUploadImage.kt:100]: Right. Thanks for the comment, this will be done in another pr where all absolutes values will be refactored.
